Source/JavaScriptCore/builtins/StringPrototype.js

```diff
@@ -226,14 +226,32 @@ function hasObservableSideEffectsForStringReplace(regexp, replacer)
     if (regexpExec !== @regExpBuiltinExec)
         return true;
 
+    var regexpFlags = @tryGetById(regexp, "flags");
```

Comment:
I wonder if we should just implement them in C++ and checking structure & watchpoints instead in the future.